build: fix missing sse42 in depends builds

theuni opened the pull request:
For depends builds without this, configure thinks that the user has overridden CXXFLAGS manually, when really they've just been set by the config.site.

The effect is that warnings and extra cxxflags (sse4.2 for example) were not being added.

laanwj commented:
But what if CXXFLAGS is overridden manually with a depends build?
(or is that not supported/possible)

ryanofsky reviewed (commented):
> But what if CXXFLAGS is overridden manually with a depends build?
(or is that not supported/possible)

It does seem to be possible:

https://github.com/bitcoin/bitcoin/blob/e222618a32a16cf0efae392e6349c10c38e57db6/depends/config.site.in#L93

Could handle this case by adding a line to config.site like `USER_CXXFLAGS="$CXXFLAGS"` and changing the condition added here to `&& test "x${USER_CXXFLAGS+set}" = x -o -n "$USER_CXXFLAGS"`

Or if this is too complicated, it would be good to add a comment explaining the ac_site_file check, since it is slightly broken.

theuni commented:
Yep, what @ryanofsky said. I was trying to keep this simple.

I think that's a good suggestion though, will do.

theuni committed:
build: always attempt to enable targeted sse42 cxxflags

This avoids a counter-intuitive drop in performance when manually adjusting the
flags.

theuni commented:
After experimenting with this for far too long, I've decided that it just doesn't make sense to tie sse42 to CXXFLAGS.

Gitian, for example, sets the CXXFLAGS manually in order to enable debugging. Linux distros will probably set them to "-O2" by default to reduce the object size. Gentoo will have some value set by the user's flags. Etc. All of those would end up with quietly disabled crc optimizations.

Instead, I think we should just enable this flags if they work, possibly behind configure flags.

For 0.15, I propose that we just keep it simple and always check for sse42, and punt worrying about the warnings until 0.16.

laanwj commented:
Not coupling sse42 to CXXFLAGS override seems like a better solution to me, too. Thanks.
utACK 9baca41
